festim/generic_simulation.py

```diff
@@ -208,13 +208,32 @@ def attribute_boundary_conditions(self):
             + [i + 1 for i, _ in enumerate(self.traps.traps)]
         )
 
+        # collect all DirichletBCs
+        dc_bcs = [
+            bc
+            for bc in self.boundary_conditions
+            if isinstance(bc, festim.DirichletBC) and (bc.field in valid_fields)
+        ]
+
         for bc in self.boundary_conditions:
             if bc.field not in valid_fields:
                 raise ValueError(f"{bc.field} is not a valid field for BC")
```

Comment:
This could me moved out of the loop into something concise like

```python
if any([bc.field not in valid_fields for bc in self.boundary_conditions]):
    raise ValueError(...)
```

Reply:
So, the error message should be reduced to `"Some fields are not valid for BC"`?